2009-09-11  Jakub Jelinek  <jakub@redhat.com>

	* combine.c (propagate_for_debug_subst): Call wrap_constant on top.

--- gcc/combine.c.jj	2009-09-08 12:32:31.000000000 +0200
+++ gcc/combine.c	2009-09-11 11:01:32.000000000 +0200
@@ -2311,7 +2311,7 @@ propagate_for_debug_subst (rtx *loc, voi
 	  to = simplify_gen_subreg (GET_MODE (x), to,
 				    GET_MODE (from), SUBREG_BYTE (x));
 	}
-      *loc = to;
+      *loc = wrap_constant (GET_MODE (x), to);
       pair->changed = true;
       return -1;
     }
